extensions/Worker.Extensions.CosmosDB/src/CosmosExtensionStartup.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+[assembly: WorkerExtensionStartup(typeof(CosmosExtensionStartup))]
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    public class CosmosExtensionStartup : WorkerExtensionStartup
+    {
+        public override void Configure(IFunctionsWorkerApplicationBuilder applicationBuilder)
+        {
+            if (applicationBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(applicationBuilder));
+            }
+
+            applicationBuilder.Services.AddAzureClientsCore(); // Adds AzureComponentFactory
+            applicationBuilder.Services.AddOptions<CosmosDBBindingOptions>();
```

Comment:
You could also add a `.Validate()` delegate to verify that _at least_ one of the two auth settings are configured correctly (conn string or AAD).